src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,198 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            linkFirst(value);
+        } else {
+            linkLast(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexToInsert(index);
+        if (index == size) {
+            add(value);
+        } else {
+            linkBefore(value, findNode(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        Node<T> antecedent = tail;
+        for (int i = 0; i < list.size(); i++) {
+            T value = list.get(i);
+            Node<T> newNode = new Node<>(antecedent, value, null);
+            if (antecedent == null) {
+                head = newNode;
+            } else {
+                antecedent.next = newNode;
+            }
+            antecedent = newNode;
+        }
+
+        tail = antecedent;
+        size += list.size();
     }
```

Comment:
When creating a new `Node<T>`, pass all the fields to the `Node` constructor. You are doing it correctly, but you should use the `linkLast` method instead of repeating the logic.